src/hotspot/share/nmt/virtualMemoryTracker.cpp

```diff
@@ -452,12 +452,12 @@ bool VirtualMemoryTracker::add_committed_region(address addr, size_t size,
   return result;
 }
 
-bool VirtualMemoryTracker::remove_uncommitted_region(address addr, size_t size) {
+bool VirtualMemoryTracker::remove_uncommitted_region(address addr, size_t size, MEMFLAGS flag) {
   assert(addr != nullptr, "Invalid address");
   assert(size > 0, "Invalid size");
   assert(_reserved_regions != nullptr, "Sanity check");
 
-  ReservedMemoryRegion  rgn(addr, size);
+  ReservedMemoryRegion  rgn(addr, size, NativeCallStack::empty_stack(), flag);
```

Comment:
Instead, change the constructor so that it takes a flag?

```c++
  ReservedMemoryRegion(address base, size_t size, MEMFLAGS flag) :
    VirtualMemoryRegion(base, size), _stack(NativeCallStack::empty_stack()), _flag(flag) { }
```

Or does that break somewhere else?

Reply:
Fixed. No problem.